### PR TITLE
move StreamJsonRpc to latest version

### PR DESCRIPTION
**Pull request by heejaechang**
latest version of JsonRpc supports custom JsonConverter which I added for these 4 types. Checksum, SolutionId, ProjectId, DocumentId

we can add more to AggregateJsonConverter for common roslyn types.

unit test added

...

this is code refactoring. there is no user experience change

### CODE REVIEW

**Comment by heejaechang** [comment]
@KevinH-MS @CyrusNajmabadi @rynowak @panopticoncentral  @mavasani @srivatsn can you take a look?

**Comment by heejaechang** [comment]
now we can use stuff like Checksum, Solution/Project/DocumentId directly in InvokeAsync in JsonRpc. as both for parameters and return type.

**Comment by rynowak** [comment]
Nice!

**Comment by heejaechang** [comment]
ping?

**Comment by KevinH-MS** [comment]
LGTM - just to confirm, there's no publicly observable change (we won't have to recompile, right)?

**Comment by heejaechang** [comment]
@KevinH-MS there is no breaking change.

**Comment by CyrusNajmabadi** [comment]
@heejaechang Is it easy for you to update the IDE features to take avantage of this?  i..e i had to write converters for DocumentId/ProjectId, can you remove those now?

**Comment by heejaechang** [comment]
@CyrusNajmabadi those serialized documentId stuff?

**Comment by CyrusNajmabadi** [comment]
yes

**Comment by heejaechang** [comment]
ping? updated

**Comment by heejaechang** [comment]
retest windows_eta_open_prtest please

**Comment by heejaechang** [comment]
@tannergooding I think the failure is real one?

**Comment by heejaechang** [comment]
@srivatsn can I get sign off for RC3? @CyrusNajmabadi are you okay with the change that affects FAR and NavigateTo? @rynowak are you okay with RazorServiceHub change?

**Comment by rynowak** [comment]
👍 from me

**Comment by tannergooding** [comment]
@heejaechang fix for the ETA failure is here: https://github.com/dotnet/roslyn-internal/pull/1447

**Comment by heejaechang** [comment]
retest windows_eta_open_prtest  please

**Comment by heejaechang** [comment]
@CyrusNajmabadi okay with FAR changes?

File: src/Workspaces/Core/Portable/Remote/RemoteArguments.cs [hunk 1]
```diff
@@ -118,14 +42,14 @@ public bool Equals(SerializableSymbolAndProjectId other)
             return new SerializableSymbolAndProjectId
             {
                 SymbolKeyData = symbolAndProjectId.Symbol.GetSymbolKey().ToString(),
-                ProjectId = SerializableProjectId.Dehydrate(symbolAndProjectId.ProjectId)
+                ProjectId = symbolAndProjectId.ProjectId
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Remote/RemoteArguments.cs]: Does ths work when the serializable type is used as aa property of another type?

**Comment by CyrusNajmabadi** [comment]
Yup.  The NavTo and FAR changes loook quite nice!

File: src/Workspaces/Core/Portable/Remote/RemoteArguments.cs (hunk 1 above)
Comment (heejaechang) [inline src/Workspaces/Core/Portable/Remote/RemoteArguments.cs]: ya. whole thing converts to json

**Pull request merged**